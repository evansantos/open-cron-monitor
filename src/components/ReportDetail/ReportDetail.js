import React, { Component } from 'react'
import CentralizedPaper from '../CentralizedPaper'
import Grid from '@material-ui/core/Grid'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

const styles = theme => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2,
  },
})

class ReportDetail extends Component {
  render() {
    const { classes } = this.props

    return (
      <CentralizedPaper>
        <Paper className={classes.root}>
          <Grid container direction="column" spacing={16}>
            <Grid item>
              <Typography>Name</Typography>
              <Typography variant="headline">{this.props.report.name}</Typography>
            </Grid>
            <Grid item>
              <Typography>Success</Typography>
              <Typography variant="headline">
                {this.props.report.success ? 'Ok' : 'Error'}
              </Typography>
            </Grid>
            <Grid item>
              <Typography>Started at</Typography>
              <Typography variant="headline">{this.props.report.startedAt}</Typography>
            </Grid>
            <Grid item>
              <Typography>Ended at</Typography>
              <Typography variant="headline">{this.props.report.endedAt}</Typography>
            </Grid>
            <Grid item>
              <Typography>Log</Typography>
              <Typography variant="headline">{this.props.report.log}</Typography>
            </Grid>
          </Grid>
        </Paper>
      </CentralizedPaper>
    )
  }
}

export default withStyles(styles)(ReportDetail)
