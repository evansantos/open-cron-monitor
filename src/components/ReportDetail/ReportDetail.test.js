import Adapter from 'enzyme-adapter-react-16'
import Enzyme from 'enzyme'
import React from 'react'
import { shallow } from 'enzyme'
import ReportDetail from './ReportDetail'

Enzyme.configure({ adapter: new Adapter() })

describe('ReportDetail', () => {
  it('should render properly', () => {
    const report = { id: 1, success: true, name: 'name-1', createdAt: 'createdAt-1', startedAt: 'startedAt-1' }
    const wrapper = shallow(<ReportDetail report={report} />)

    expect(wrapper.dive()).toMatchSnapshot();
  })
})
