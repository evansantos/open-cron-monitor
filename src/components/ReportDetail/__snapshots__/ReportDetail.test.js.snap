// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDetail should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportDetail
    classes={
        Object {
            "root": "ReportDetail-root-1",
          }
    }
    report={
        Object {
            "createdAt": "createdAt-1",
            "id": 1,
            "name": "name-1",
            "success": true,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Paper)
        className="ReportDetail-root-1"
>
        <WithStyles(Grid)
                container={true}
                direction="column"
                spacing={16}
        >
                <WithStyles(Grid)
                        item={true}
                >
                        <WithStyles(Typography)>
                                Name
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                variant="headline"
                        >
                                name-1
                        </WithStyles(Typography)>
                </WithStyles(Grid)>
                <WithStyles(Grid)
                        item={true}
                >
                        <WithStyles(Typography)>
                                Success
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                variant="headline"
                        >
                                Ok
                        </WithStyles(Typography)>
                </WithStyles(Grid)>
                <WithStyles(Grid)
                        item={true}
                >
                        <WithStyles(Typography)>
                                Created at
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                variant="headline"
                        >
                                createdAt-1
                        </WithStyles(Typography)>
                </WithStyles(Grid)>
                <WithStyles(Grid)
                        item={true}
                >
                        <WithStyles(Typography)>
                                Log
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                variant="headline"
                        />
                </WithStyles(Grid)>
        </WithStyles(Grid)>
</WithStyles(Paper)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Grid)
          container={true}
          direction="column"
          spacing={16}
>
          <WithStyles(Grid)
                    item={true}
          >
                    <WithStyles(Typography)>
                              Name
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                              variant="headline"
                    >
                              name-1
                    </WithStyles(Typography)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
                    item={true}
          >
                    <WithStyles(Typography)>
                              Success
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                              variant="headline"
                    >
                              Ok
                    </WithStyles(Typography)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
                    item={true}
          >
                    <WithStyles(Typography)>
                              Created at
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                              variant="headline"
                    >
                              createdAt-1
                    </WithStyles(Typography)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
                    item={true}
          >
                    <WithStyles(Typography)>
                              Log
                    </WithStyles(Typography)>
                    <WithStyles(Typography)
                              variant="headline"
                    />
          </WithStyles(Grid)>
</WithStyles(Grid)>,
        "className": "ReportDetail-root-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Grid)
              item={true}
>
              <WithStyles(Typography)>
                            Name
              </WithStyles(Typography)>
              <WithStyles(Typography)
                            variant="headline"
              >
                            name-1
              </WithStyles(Typography)>
</WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
>
              <WithStyles(Typography)>
                            Success
              </WithStyles(Typography)>
              <WithStyles(Typography)
                            variant="headline"
              >
                            Ok
              </WithStyles(Typography)>
</WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
>
              <WithStyles(Typography)>
                            Created at
              </WithStyles(Typography)>
              <WithStyles(Typography)
                            variant="headline"
              >
                            createdAt-1
              </WithStyles(Typography)>
</WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
>
              <WithStyles(Typography)>
                            Log
              </WithStyles(Typography)>
              <WithStyles(Typography)
                            variant="headline"
              />
</WithStyles(Grid)>,
          ],
          "container": true,
          "direction": "column",
          "spacing": 16,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)>
                  Name
</WithStyles(Typography)>,
                <WithStyles(Typography)
                  variant="headline"
>
                  name-1
</WithStyles(Typography)>,
              ],
              "item": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Name",
                },
                "ref": null,
                "rendered": "Name",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "name-1",
                  "variant": "headline",
                },
                "ref": null,
                "rendered": "name-1",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)>
                  Success
</WithStyles(Typography)>,
                <WithStyles(Typography)
                  variant="headline"
>
                  Ok
</WithStyles(Typography)>,
              ],
              "item": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Success",
                },
                "ref": null,
                "rendered": "Success",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Ok",
                  "variant": "headline",
                },
                "ref": null,
                "rendered": "Ok",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)>
                  Created at
</WithStyles(Typography)>,
                <WithStyles(Typography)
                  variant="headline"
>
                  createdAt-1
</WithStyles(Typography)>,
              ],
              "item": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Created at",
                },
                "ref": null,
                "rendered": "Created at",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "createdAt-1",
                  "variant": "headline",
                },
                "ref": null,
                "rendered": "createdAt-1",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)>
                  Log
</WithStyles(Typography)>,
                <WithStyles(Typography)
                  variant="headline"
/>,
              ],
              "item": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Log",
                },
                "ref": null,
                "rendered": "Log",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                  "variant": "headline",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Paper)
          className="ReportDetail-root-1"
>
          <WithStyles(Grid)
                    container={true}
                    direction="column"
                    spacing={16}
          >
                    <WithStyles(Grid)
                              item={true}
                    >
                              <WithStyles(Typography)>
                                        Name
                              </WithStyles(Typography)>
                              <WithStyles(Typography)
                                        variant="headline"
                              >
                                        name-1
                              </WithStyles(Typography)>
                    </WithStyles(Grid)>
                    <WithStyles(Grid)
                              item={true}
                    >
                              <WithStyles(Typography)>
                                        Success
                              </WithStyles(Typography)>
                              <WithStyles(Typography)
                                        variant="headline"
                              >
                                        Ok
                              </WithStyles(Typography)>
                    </WithStyles(Grid)>
                    <WithStyles(Grid)
                              item={true}
                    >
                              <WithStyles(Typography)>
                                        Created at
                              </WithStyles(Typography)>
                              <WithStyles(Typography)
                                        variant="headline"
                              >
                                        createdAt-1
                              </WithStyles(Typography)>
                    </WithStyles(Grid)>
                    <WithStyles(Grid)
                              item={true}
                    >
                              <WithStyles(Typography)>
                                        Log
                              </WithStyles(Typography)>
                              <WithStyles(Typography)
                                        variant="headline"
                              />
                    </WithStyles(Grid)>
          </WithStyles(Grid)>
</WithStyles(Paper)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Grid)
            container={true}
            direction="column"
            spacing={16}
>
            <WithStyles(Grid)
                        item={true}
            >
                        <WithStyles(Typography)>
                                    Name
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                    variant="headline"
                        >
                                    name-1
                        </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
                        item={true}
            >
                        <WithStyles(Typography)>
                                    Success
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                    variant="headline"
                        >
                                    Ok
                        </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
                        item={true}
            >
                        <WithStyles(Typography)>
                                    Created at
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                    variant="headline"
                        >
                                    createdAt-1
                        </WithStyles(Typography)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
                        item={true}
            >
                        <WithStyles(Typography)>
                                    Log
                        </WithStyles(Typography)>
                        <WithStyles(Typography)
                                    variant="headline"
                        />
            </WithStyles(Grid)>
</WithStyles(Grid)>,
          "className": "ReportDetail-root-1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Grid)
                item={true}
>
                <WithStyles(Typography)>
                                Name
                </WithStyles(Typography)>
                <WithStyles(Typography)
                                variant="headline"
                >
                                name-1
                </WithStyles(Typography)>
</WithStyles(Grid)>,
              <WithStyles(Grid)
                item={true}
>
                <WithStyles(Typography)>
                                Success
                </WithStyles(Typography)>
                <WithStyles(Typography)
                                variant="headline"
                >
                                Ok
                </WithStyles(Typography)>
</WithStyles(Grid)>,
              <WithStyles(Grid)
                item={true}
>
                <WithStyles(Typography)>
                                Created at
                </WithStyles(Typography)>
                <WithStyles(Typography)
                                variant="headline"
                >
                                createdAt-1
                </WithStyles(Typography)>
</WithStyles(Grid)>,
              <WithStyles(Grid)
                item={true}
>
                <WithStyles(Typography)>
                                Log
                </WithStyles(Typography)>
                <WithStyles(Typography)
                                variant="headline"
                />
</WithStyles(Grid)>,
            ],
            "container": true,
            "direction": "column",
            "spacing": 16,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Typography)>
                    Name
</WithStyles(Typography)>,
                  <WithStyles(Typography)
                    variant="headline"
>
                    name-1
</WithStyles(Typography)>,
                ],
                "item": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Name",
                  },
                  "ref": null,
                  "rendered": "Name",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "name-1",
                    "variant": "headline",
                  },
                  "ref": null,
                  "rendered": "name-1",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Typography)>
                    Success
</WithStyles(Typography)>,
                  <WithStyles(Typography)
                    variant="headline"
>
                    Ok
</WithStyles(Typography)>,
                ],
                "item": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Success",
                  },
                  "ref": null,
                  "rendered": "Success",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Ok",
                    "variant": "headline",
                  },
                  "ref": null,
                  "rendered": "Ok",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Typography)>
                    Created at
</WithStyles(Typography)>,
                  <WithStyles(Typography)
                    variant="headline"
>
                    createdAt-1
</WithStyles(Typography)>,
                ],
                "item": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Created at",
                  },
                  "ref": null,
                  "rendered": "Created at",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "createdAt-1",
                    "variant": "headline",
                  },
                  "ref": null,
                  "rendered": "createdAt-1",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Typography)>
                    Log
</WithStyles(Typography)>,
                  <WithStyles(Typography)
                    variant="headline"
/>,
                ],
                "item": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Log",
                  },
                  "ref": null,
                  "rendered": "Log",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": undefined,
                    "variant": "headline",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
